highway.tex

```diff
@@ -24,807 +26,1323 @@
-When $V$ receives endorsements for any vote $\mu$ by a set of validators comprising at least $2/3$rds of the total validator weight (i.e. $V$ receives confirmation that $\mu$ is endorsed), $V$ will remove $\mu$ and all votes in its justification from their verification buffer.
 
-We now need to show that this strategy satisfies our requirements. First, note that since $V$ will only endorse votes by senders not known to be equivocators, $V$ will never endorse a pair of equivocating votes. This is because upon initial receipt of the second vote, $V$ will know the sender to be an equivocator and thus will not endorse. On the other hand, an honest validator will never equivocate and thus will never be on any other honest validator's list of equivocators. Thus, if endorsements are solicited for a vote $\mu$ sent by an honest validator $V$, all other honest validators will eventually endorse. Assuming that these honest validators comprise at least $2/3$rds of the total validator weight, this means that $\mu$ will eventually be endorsed. Furthermore, during periods of network stability, this will happen in time at most $O(\Delta)$. Thus, if an honest validator $V$ sends a vote $\mu$ to another honest validator $W$, it will take at most a constant number of rounds of communication for $W$ to verify the validity of the vote and at most a constant number of rounds to endorse it (if necessary), by which point $W$ will be ready to include $\mu$ in their next vote.
 
-We have left to show that an honest validator will only send messages satisfying Limited Naivety Criterion. This is because any message that clears $V$'s verification buffer either:
+\desc{
+
+\begin{center}
+    {\bf Round Exponent Maintenance Strategy for $V\in \cV$} 
+\end{center}
+% \damian{I wrote a slightly simpler version than what was written here before. I don't think it matters much how this is done. Feel free to roll it back to the previous version. }
+\begin{enumerate}
+\item At tick $i=0$, initialize $n_v(j)= \nmin$ for all $j$. Initialize $\cntsucc=0$.
+\item At tick $i>0$,  let $m:=n_v(i)$. Whenever $i$ is divisible by $2^{m+1}$:
+% \item At tick $i>0$,  let $m:=n_v(i)$. Whenever $i$ is divisible by $2^{m+1}$ and $n_v$ did not change in $C$ previous rounds (i.e., it was constant in the interval $[i-C2^{m+1}, i-1]$):
 \begin{enumerate}
-\item Was initially put into the verification buffer, but was (or was cited by) a vote that was later endorsed.
-\item Was sent by a validator who was not at the time in $E_V$.
+        \item Let $b_{\mathrm{fin}}$ denote the number of blocks finalized with confidence $t_0$ during the last $C$ rounds,
+        \item if $b_{\mathrm{fin}}<C_{\mathrm{fail}}$ then set $n_v(j)=\min(m+1, \nmax)$, for all $j\geq i$,
+        \item if $b_{\mathrm{fin}}>C_{\mathrm{succ}}$ then increment $\cntsucc=\cntsucc+1$, otherwise set $\cntsucc=0$.
```

Comment:
Both of these could be non-strict inequalities, since above we already require that C<sub>fail</sub> is _strictly_ less than C<sub>succ</sub>.